### PR TITLE
Add loan limitter to TrueLender

### PR DESCRIPTION
**Pull request by MiksuJak**

### CODE REVIEW

File: contracts/truefi/TrueLender.sol [hunk 1]
```diff
@@ -74,6 +74,11 @@ contract TrueLender is ITrueLender, Ownable {
     // minimum prediction market voting period
     uint256 public votingPeriod;
 
+    // ===== End of pool parameters =====
```
Comment (vanruch) [inline contracts/truefi/TrueLender.sol:77]: I don't get this comment, isn;t maxLoans also a pool parameter?

File: contracts/truefi/TrueLender.sol [hunk 2]
```diff
@@ -122,6 +127,12 @@ contract TrueLender is ITrueLender, Ownable {
      */
     event TermLimitsChanged(uint256 minTerm, uint256 maxTerm);
 
+    /**
+     * @dev Emitted when loans limit is change
+     * @param maxLoans new maximum amount of loans
+     */
+    event LoansLimitChanged(uint256 maxLoans);
```
Comment (vanruch) [inline contracts/truefi/TrueLender.sol:134]: 👍

File: contracts/truefi/TrueLender.sol [hunk 3]
```diff
@@ -240,6 +253,16 @@ contract TrueLender is ITrueLender, Ownable {
         emit RiskAversionChanged(newRiskAversion);
     }
 
+    /**
+     * @dev Set new loans limit. Only owner can change parameters.
+     * @param newLoansLimit New loans limit
+     */
+    function setLoansLimit(uint256 newLoansLimit) external onlyOwner {
+        require(newLoansLimit >= _loans.length, "TrueLender: Cannot set loans amount to less than active loans");
```
Comment (vanruch) [inline contracts/truefi/TrueLender.sol:261]: Not sure about this require - in theory we might want to change it to a value lower and don't accept any loans till old ones expire

Comment (vanruch) [inline contracts/truefi/TrueLender.sol:261]: Not sure about this require - in theory we might want to change it to a value lower and don't accept any loans till old ones expire

File: contracts/truefi/TrueLender.sol (hunk 1 above)
Comment (MiksuJak) [reply contracts/truefi/TrueLender.sol:77]: Very good point. I probably tried to overdo it. Will fix this

File: contracts/truefi/TrueLender.sol (hunk 3 above)
Comment (vanruch) [reply contracts/truefi/TrueLender.sol:261]: Not sure if it will ever happen, but it doesn't hurt us to leave this option

Comment (MiksuJak) [reply contracts/truefi/TrueLender.sol:261]: Sure. Will remove it

Comment (MiksuJak) [reply contracts/truefi/TrueLender.sol:261]: Do You mean setter or require. Ive got a little confused there

File: contracts/truefi/TrueLender.sol [hunk 4]
```diff
@@ -265,6 +288,7 @@ contract TrueLender is ITrueLender, Ownable {
     function fund(ILoanToken loanToken) external onlyAllowedBorrowers {
         require(loanToken.isLoanToken(), "TrueLender: Only LoanTokens can be funded");
         require(loanToken.currencyToken() == currencyToken, "TrueLender: Only the same currency LoanTokens can be funded");
+        require(_loans.length < maxLoans, "TrueLender: Loans amount has reached the limit");
```
Comment (barrutko) [inline contracts/truefi/TrueLender.sol:291]:
```suggestion
        require(_loans.length < maxLoans, "TrueLender: Loans number has reached the limit");
```

File: test/truefi/TrueLender.test.ts [hunk 5]
```diff
@@ -262,6 +286,12 @@ describe('TrueLender', () => {
         .to.be.revertedWith('TrueLender: Sender is not allowed to borrow')
     })
 
+    it('reverts if loan amount would exceed the limit', async () => {
+      await lender.setLoansLimit(0)
+      await expect(lender.fund(mockLoanToken.address))
+        .to.be.revertedWith('TrueLender: Loans amount has reached the limit')
```
Comment (barrutko) [inline test/truefi/TrueLender.test.ts:292]:
```suggestion
        .to.be.revertedWith('TrueLender: Loans number has reached the limit')
```

File: contracts/truefi/TrueLender.sol (hunk 3 above)
Comment (vanruch) [reply contracts/truefi/TrueLender.sol:261]: require

**Review by vanruch — APPROVED** [review]

**Pull request merged**